tests/unit/proxy_configuration/test_tiers.py

```diff
@@ -94,3 +94,43 @@ async def test_successful_request_makes_tier_go_down() -> None:
 
     assert info is not None
     assert info.url == tiered_proxy_urls[0][0]
+
+
+async def test_none_proxy_retrying_request_makes_tier_go_up() -> None:
+    tiered_proxy_urls: list[list[str | None]] = [
+        [None],
+        ['http://proxy:1111'],
+    ]
+
+    config = ProxyConfiguration(tiered_proxy_urls=tiered_proxy_urls)
+
+    # Calling `new_proxy_info` with the same request most probably means it's being retried
+    request_1 = Request(url='http://some.domain/abc', unique_key='1', id='1')
+
+    # No proxy used.
+    info = await config.new_proxy_info(None, request_1, None)
+    assert info is None
+
+    info = await config.new_proxy_info(None, request_1, None)
```

Comment:
Could you add a comment here explaining what is expected to happen? e.g., on the second call of the same request, the tier increases

(it can be either a comment or assert message)

Reply:
Done